packages/chain-adapters/src/evm/EvmBaseAdapter.ts

```diff
@@ -269,26 +270,43 @@ export abstract class EvmBaseAdapter<T extends EvmChainId> implements IChainAdap
           }
         }
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return { gasPrice: numberToHex(tx.chainSpecific.gasPrice!) }
+        return { gasPrice: numberToHex(gasPrice!) }
       })()
 
       const bip44Params = this.getBIP44Params({ accountNumber })
       const txToSign = {
         addressNList: toAddressNList(bip44Params),
-        value: numberToHex(isTokenSend ? '0' : tx.value),
+        value: numberToHex(isTokenSend ? '0' : value),
         to: destAddress,
         chainId: Number(fromChainId(this.chainId).chainReference),
         data,
         nonce: numberToHex(account.chainSpecific.nonce),
         gasLimit: numberToHex(gasLimit),
         ...fees,
       } as SignTx<T>
-      return { txToSign }
+      return txToSign
     } catch (err) {
       return ErrorHandler(err)
     }
   }
 
+  async buildSendTransaction(input: BuildSendTxInput<T>): Promise<{
+    txToSign: SignTx<T>
+  }> {
+    if (!this.supportsChain(input.wallet)) {
+      throw new Error(`wallet does not support ${this.getDisplayName()}`)
+    }
+
+    await this.assertSwitchChain(input.wallet)
+
+    const from = await this.getAddress(input)
+    const txToSign = await this.buildSignTx({
+      ...input,
+      from,
+    })
+    return { txToSign }
+  }
+
```

Comment:
changes to this file are essentially the following:
- keep backwards compatibility for `buildSendTransaction` so prod can keep working
- create `buildSignTx` to allow tx build without wallet
- expose `assertSwitchChain` to allow caller to switch chain